model-optimizer/unit_tests/mock_mo_frontend/mock_mo_ngraph_frontend/mock_mo_frontend.hpp

```diff
@@ -149,7 +163,28 @@ class MOCK_API PlaceMockPy : public Place
     {
         m_stat.m_is_equal++;
         m_stat.m_lastArgPlace = another;
-        return m_name == another->get_names().at(0);
+        std::shared_ptr<PlaceMockPy> mock = std::dynamic_pointer_cast<PlaceMockPy>(another);
+        return m_name == mock->m_name && m_is_op == mock->m_is_op &&
+               m_portIndex == mock->m_portIndex;
+    }
+
+    bool is_equal_data(Ptr another) const override
+    {
+        m_stat.m_is_equal_data++;
+        m_stat.m_lastArgPlace = another;
+        std::shared_ptr<PlaceMockPy> mock = std::dynamic_pointer_cast<PlaceMockPy>(another);
+        if (mock->m_name.find("conv2d") != std::string::npos &&
+            m_name.find("conv2d") != std::string::npos)
+        {
+            return true;
+        }
+        if ((mock->m_name.find("8") != std::string::npos ||
+             mock->m_name.find("9") != std::string::npos) &&
+            (m_name.find("8") != std::string::npos || m_name.find("9") != std::string::npos))
+        {
+            return true;
+        }
```

Comment:
Hardcoding values in the mock test infrastructure looks ugly. Please, consider refactoring.

Reply:
Done, extended "mock API" to set these values from test code